common/transaction/types.go

```diff
@@ -23,6 +24,13 @@ type Validity struct {
 }
 
 type ValidTransaction struct {
-	// extrinsic common.Extrinsic
-	validity Validity
+	extrinsic types.Extrinsic
+	validity  *Validity
```

Comment:
I feel like this shouldn't be a pointer as the extrinsic is not. Thoughts?

Reply:
the validity is a struct and structs are generally pointers? cause when we do `new(Validity)` it's a pointer

Reply:
:confused: the reason to use a pointer or not should typically depend on whether you want a copy or a shared instance. In this case you are (re-)allocating memory for `extrinsic` but referencing a (possibly) shared instance of `validity`. Generally if we don't want any chance of mutation we should not use pointers, on the other hand if we want to be space efficient we should.

Reply:
okay I'll take a look. I think we should probably use pointers but I'll dig a bit deeper

Reply:
it's a bit weird to use types.Extrinsic as a pointer since its underlying type is `[]byte` but we can't cast `*[]byte` to `*types.Extrinsic` and vice versa, but we can for `[]byte` and `types.Extrinsic`. However I've updated the code if you want to take a look at the changes.

Reply:
I think the solution there is to deref and then cast? Looks good :+1:

Reply:
yeah it does work, just an extra step lol. cool